.github/workflows/tests.yml

```diff
@@ -0,0 +1,74 @@
+name: Run all tests
+
+# Run this workflow every time a new commit is pushed to the repository
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  run-all-tests:
+    name: Run all tests
+
+    # TODO: not sure what the best OS to run on is so ubuntu will do for now
```

Comment:
Ubuntu is fine; nothing in this track should be OS-specific enough that the host OS actually matters.